src/main/java/com/cloudbees/jenkins/support/configfiles/SecretHandler.java

```diff
@@ -90,15 +92,17 @@ public void characters(char[] ch, int start, int length) throws SAXException {
             }
         };
         String str = FileUtils.readFileToString(xmlFile);
-        Source src = new SAXSource(xr, new InputSource(new StringReader(str)));
         final ByteArrayOutputStream result = new ByteArrayOutputStream();
         Result res = new StreamResult(result);
-        Transformer transformer = TransformerFactory.newInstance().newTransformer();
+        TransformerFactory factory = TransformerFactory.newInstance();
+        factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+        Transformer transformer = factory.newTransformer();
         //omit xml declaration because of https://bugs.openjdk.java.net/browse/JDK-8035437
         transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
         transformer.setOutputProperty(OutputKeys.ENCODING, OUTPUT_ENCODING);
 
         try {
+            Source src = createSafeSource(xr, new InputSource(new StringReader(str)));
```

Comment:
Should we move this out of the try/catch so that if `createSafeSource` throws a `TransformerException` (should only happen if the underlying XML lib doesn't support the `http://xml.org/sax/features/external-*-entities` features and the file has an XXE) and is unable to remove the external entities it won't be added to the bundle? Currently the file would be added with the XXE intact but unexecuted.

Reply:
Ideally we want to give the support a bundle with as much information as possible, especially if there are some potential attacks in place. But OTOH we do not want some automatic parsing tool from the support to trigger the attack and so become vulnerable to support bundle. From my PoV the best approach if we know there is a XXE in a XML, to put inside the bundle but either with its content commented out or the extension changed to something like `.dangerous_xml`.

Otherwise, not inserting the file seems a safer approach. We do not know for what usage this bundle could be and so better safe than sorry.